static/js/message_util.js

```diff
@@ -20,8 +20,14 @@ function add_messages(messages, msg_list, opts) {
 exports.add_old_messages = function (messages, msg_list) {
     return add_messages(messages, msg_list, {messages_are_new: false});
 };
-exports.add_new_messages = function (messages, msg_list) {
-    return add_messages(messages, msg_list, {messages_are_new: true});
+
+exports.add_new_messages = function (messages, msg_list, opts) {
+    if (opts) {
+        opts.messages_are_new = true;
+    } else {
+        opts = {messages_are_new: true};
+    }
```

Comment:
We should do this using a modern ES6 pattern for default arguments, it'd be a lot cleaner.